### DISCUSSION TITLE
how do I configure video size in Birdseye?

### DISCUSSION THREAD

**Discussion by tedder** [blakeblackshear/frigate#9544] [Ask A Question]
My Birdseye layout has always looked strange, but going from 13-rc to 13.0 it got much worse. I have 8 cameras of varying resolutions, it's always been crowded near the top of the page but it got significantly worse. I've attached a screenshot showing the layout.

<img width="1380" alt="Screenshot 2024-01-31 at 3 07 44 PM" src="https://github.com/blakeblackshear/frigate/assets/97914/3cacd7ad-d915-4ce6-ba52-b759f2097dab">

In the docs there's nothing specific about sizing in Birdseye. The only resolution settings I have are in `detect:` sections. I feel like there used to be some settings in `inputs`, but I may be wrong. Where should I look?

For the record, here's my birdseye config section:
```
birdseye:
  enabled: true
  mode: continuous # motion,objects
```

**Comment by NickM-27** [comment]
There were no changes between 13-RC and 13

The size of birdseye is configurable https://docs.frigate.video/configuration/reference

**Reply by tedder** [reply]
I'm not talking about the overall size, I'm talking about how the cameras "pack" inside of it. Ideally it would nearly fill the birdseye window, rather than being in the top 20-30%.

**Reply by NickM-27** [reply]
Birdseye uses a lot of logic to decide how to best pack the cameras into the view. It was unable to find a layout that worked with that many cameras 

Changing the resolution of birdseye does affect this in that changing the aspect ratio affects how things are packed. The order of cameras also matters. It is generally best to order cameras from widest to tallest aspect ratio.

**Reply by tedder** [reply]
great. I'll try setting the ordering and use widest->tallest.

**Reply by f1d094** [reply]
@NickM-27 - Would the ordering recommendation be the same for a 9:16 screen?

The screen I am wanting to fill is 1080w x 1920h and 5 of 21 cameras similarly in 9:16 screen ratio. Of the rest, 8 are 16:9 and 8 are 4:3. In birdseye, they tend to pack in up to 4 across the top and rather small rather than fill the available space. I tried:
 ```
height: 1920
width: 1080
layout:
  scaling_factor: 1
``` 
That didn't seem to have significant effect other than to cause some complaints on startup `ERROR   : Error finding appropriate birdseye layout`

Are there any other knobs I can turn? I'm running 0.13.2-6476f8a

**Reply by f1d094** [reply]
@NickM-27: Nevermind. Sorting the cameras so that similar sizes are ordered together made a big improvement. Works much more as expected now.
